src/Razor/src/Microsoft.AspNetCore.Razor.ProjectEngineHost/Utilities/StreamExtensions.NetCore.cs

```diff
@@ -73,9 +74,15 @@ public static Task<string> ReadProjectInfoRemovalAsync(this Stream stream, Cance
         return stream.ReadStringAsync(encoding: null, cancellationToken);
     }
 
+    public static async Task WriteWithSizeAsync<T>(this Stream stream, T value, CancellationToken cancellationToken)
+    {
+        var bytes = RazorMessagePackSerializer.Serialize(value);
+        WriteSize(stream, bytes.Length);
+        await stream.WriteAsync(bytes, 0, bytes.Length, cancellationToken).ConfigureAwait(false);
+    }
     public static async Task WriteProjectInfoAsync(this Stream stream, RazorProjectInfo projectInfo, CancellationToken cancellationToken)
     {
-        var bytes = projectInfo.Serialize();
+        var bytes = RazorMessagePackSerializer.Serialize(projectInfo);
         WriteSize(stream, bytes.Length);
         await stream.WriteAsync(bytes, 0, bytes.Length, cancellationToken).ConfigureAwait(false);
```

Comment:
I'm guessing that you're planning this for a later change, but consider calling `SerializeToAsync(Stream, T, CancellationToken)` that takes a stream.